omniduct/filesystems/s3.py

```diff
@@ -189,11 +189,27 @@ def _dir(self, path):
     #         if pattern is None or pattern.match(k.key[len(path_prefix):]):
     #             yield k.key
 
-    def _mkdir(self, path, recursive):
-        raise NotImplementedError("Support for S3 write operations has yet to be implemented.")
+    def _mkdir(self, path, recursive, exist_ok):
+        if not path.endswith('/'):
+            path += '/'
+        if not self._exists(path):
+            self._client.put_object(Bucket=self.bucket, Key=self._path(path))
+
+    def _remove_file(self, path):
```

Comment:
Let's make this is a private method: `def __remove_file(...)`.

Reply:
adding a condition to check self.isdir(path) and if it's not directory then will call  __remove_one.

Reply:
adding the conditions to deal even the user passes the file and recursive as True.